cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/mazha/CouchClient.java

```diff
@@ -452,12 +453,17 @@ public ChangesResult changes(Object since, Integer limit) {
     public ChangesResult changes(PullFilter filter, Object since, Integer limit) {
         Map<String, Object> options = getParametrizedChangeFeedOptions(since, limit);
         if (filter != null) {
-            String filterName = filter.getName();
-            Map filterParameters = filter.getParameters();
-            if (filterName != null) {
-                options.put("filter", filterName);
-                if (filterParameters != null) {
-                    options.putAll(filterParameters);
+            if (filter instanceof PullDocIdsFilter) {
```

Comment:
I don't think it's a good idea to make a special case for this. We can `POST` the options in the body for all `changes` requests

Reply:
Ok, checking again the API doc, I see that POST could be used also for standard filters in the _changes endpoint